google-cloud-spanner/lib/google/cloud/spanner/commit.rb

```diff
@@ -127,6 +129,8 @@ def upsert table, *rows
         #   See [Data
         #   types](https://cloud.google.com/spanner/docs/data-definition-language#data_types).
         #
+        # @return [Array<Hash>] The array of rows.
```

Comment:
I think it would be better to leave that behavior undocumented. It's not part of the API design.